### PR TITLE
previous and next buttons added on learn pages

### LINKED ISSUE #4902: Add Next/Previous option on Learn pages
## Description
It will provide easier navigation between the Learn pages if we can provide a next/previous option[1] as shown below.

Screenshot 2022-07-27 at 14 37 31

[1] https://wso2.com/choreo/docs/get-started/what-is-choreo/

### PR DESCRIPTION
**Pull request by s-vamshi**
> Fixes #4902 

This is how it looks.
image

**NOTE:** This are not final changes. There are few changes to be made.

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Comment by s-vamshi** [comment]
@sm1990 I have added button. Can you please check if they are working? And if I am going in the right direction?
There few things which needs to be changed like styles, which I will do subsequently.

File: components/common/prev-next/PrevNext.js [hunk 1]
```diff
@@ -20,89 +20,115 @@ import * as React from 'react';
 import { Col } from 'react-bootstrap';
 
 
-import LearnToc from '../../../files1.json';
 import { prefix } from '../../../utils/prefix';
+import { useState } from 'react';
```
Comment (sm1990) [inline components/common/prev-next/PrevNext.js:24]: Move this to line 21. We usually keep all the external packages/libraries at the top.

**Comment by sm1990** [comment]
@s-vamshi seems like you are on the right track. I noticed that the links don't work in `Build and run` section. Also for styling, you can refer https://ballerina.io/learn/by-example/hello-world-service/

**Comment by s-vamshi** [comment]
@sm1990 will this pr get counted towards Hacktoberfest if it gets merged tomorrow?

**Comment by sm1990** [comment]
> @sm1990 will this pr get counted towards Hacktoberfest swags if it gets merged tomorrow?

Yes. This will be still considered. :smile:

**Comment by s-vamshi** [comment]
@sm1990 great! If we are in first page, should we show the previous button which links to the last page and vice versa?

**Comment by s-vamshi** [comment]
id: cli-commands is same for two objects in learn-lm.json thats the reason in build run section cli command is linking wrongly

**Comment by sm1990** [comment]
> @sm1990 great! If we are in first page, should we show the previous button which links to the last page and vice versa?

The first page, should not have a previous button.

**Comment by s-vamshi** [comment]
Yes can you please review once?

**Comment by sm1990** [comment]
> cli-commands
The whole `reference` section seems to be a leftover from the last restructuring we did. You can remove the whole object.(
https://github.com/ballerina-platform/ballerina-dev-website/blob/c87708614316ebbfd7e1ebd686638be8f6bf3743/utils/learn-lm.json#L1260)

**Comment by sm1990** [comment]
@s-vamshi increase the space between the buttons and the text.
image
image

**Comment by s-vamshi** [comment]
@sm1990 should I remove references from learn lm .json

**Comment by sm1990** [comment]
> @sm1990 should I remove references from learn lm .json

yes

**Comment by sm1990** [comment]
@s-vamshi can you make the links turn to dark grey (refer the `Back to pre-built` button in https://ballerina.io/learn/pre-built-integrations/google-sheets-to-salesforce-integration/) on hover. Also remove the underline.

**Comment by s-vamshi** [comment]
@sm1990 sure will do that

**Comment by s-vamshi** [comment]
@sm1990 I didn't submit swags form for this PR as it has not been merged, in discord there is an announcement that yesterday was the last date

**Comment by s-vamshi** [comment]
@sm1990 Can you please review now once?

**Comment by s-vamshi** [comment]
@sm1990 Can you please review changes?

**Comment by sm1990** [comment]
> @sm1990 I didn't submit swags form for this PR as it has not been merged, in discord there is an announcement, should I submit with PR link even if not merged?

Yes you can submit the pr link. We will keep a note :smile:

**Comment by sm1990** [comment]
Getting this error when I click on the next link.
image

**Comment by s-vamshi** [comment]
@sm1990 Can you please review now?
in learn lm.json there are two objects with id **coding-conventions** so removed the object which is not been used

File: components/common/prev-next/PrevNext.js [hunk 2]
```diff
@@ -17,92 +17,266 @@
...
+import { Row,Col } from 'react-bootstrap';
+import Link from "next/link";
+
 
 
-import LearnToc from '../../../files1.json';
 import { prefix } from '../../../utils/prefix';
+
 // import styles from './LeftNav.module.css';
```
Comment (sm1990) [inline components/common/prev-next/PrevNext.js:28]: Remove unused code

Comment (s-vamshi) [reply components/common/prev-next/PrevNext.js]: Done

**Comment by sm1990** [comment]
@s-vamshi all seems to be good. But I noticed something with the flow. 
In `Get started with integration` section we have couple of pages opening in new tabs (e.g. [Pre-built integrations](https://pre-prod.ballerina.io/learn/pre-built-integrations/), [Enterprise integration patterns](https://pre-prod.ballerina.io/learn/enterprise-integration-patterns/), and [Integration tutorials](https://pre-prod.ballerina.io/learn/integration-tutorials)). 

Can we have those links opening in new tabs to maintain consistency? There are few more such links. you can find them in the `Learn` landing page https://ballerina.io/learn/.

**Comment by s-vamshi** [comment]
Sure will do that

**Comment by s-vamshi** [comment]
@sm1990 can you please check now?

**Comment by s-vamshi** [comment]
@sm1990 can you please check now?

**Review by sm1990 — APPROVED** [review]

**Comment by sm1990** [comment]
> @sm1990 can you please check now?

@s-vamshi Thank you very mush for your contribution. Hope you filled the form in https://ballerina.io/hacktoberfest/#rules.

**Comment by s-vamshi** [comment]
> > @sm1990 can you please check now?
> 
> @s-vamshi Thank you very mush for your contribution. Hope you filled the form in https://ballerina.io/hacktoberfest/#rules.
😃
Yes I have filled the form last week when PR was not merged

**Pull request merged**